srecord/arglex.cc

```diff
@@ -277,7 +277,7 @@ srecord::arglex::compare(const char *formal, const char *actual) -> bool
 static auto
 is_a_number(const char *s, long &n) -> int
 {
-    int         sign;
+    int         sign = 0;
```

Comment:
Yes to the check and the fixes. In most cases, variables are eventually initialised, but I agree with this one for the preventative value alone.